### PR TITLE
feat: [CHK-3253] now common supports on NPG client language parameter

### PR DESCRIPTION
**Pull request by EmanueleBVtech**
#### List of Changes
- added a new method to the NPG client with provides access to the language selector
- if language is 2bytes (it, en) -> it gets converted to 3bytes (ITA, ENG, etc...)

#### Motivation and Context
This change is part of a bigger task aiming to provide NPG with the language preference as set by the user on the checkout FE.

#### Types of changes
Unit test

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java [hunk 1]
```diff
@@ -321,10 +323,71 @@ public Mono<FieldsDto> buildForm(
...
+                                     @NotNull URI merchantUrl,
+                                     @NotNull URI resultUrl,
+                                     @NotNull URI notificationUrl,
+                                     @NotNull URI cancelUrl,
+                                     @NotNull String orderId,
+                                     @NotNull String customerId,
+                                     @NonNull PaymentMethod paymentMethod,
+                                     @NonNull String defaultApiKey,
+                                     String contractId,
+                                     String language
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java:373]: Add the `@Nullable` annotation if they can be `null` :)

```suggestion
                                     @Nullable String contractId,
                                     @Nullable String language
```

File: src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java [hunk 2]
```diff
@@ -644,8 +710,34 @@ private CreateHostedOrderRequestDto buildOrderRequestDto(
...
+                    ISO_639_3_lang = "deu";
+                    break;
+                case "sl":
+                    ISO_639_3_lang = "slv";
+                    break;
+                default:
+                    ISO_639_3_lang = "ita";
+                    break;
+            }
+        }
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java:739]: We could use a static `Map` instead, what do you think?

```java
private static final LANG_MAP = Map.of(
    "it", "ita",
    "fr", "fra",
    "de", "deu"
);
```

and then

```suggestion
        if (language != null) {
            ISO_639_3_lang = LANG_MAP.getOrDefault(language, CREATE_HOSTED_ORDER_REQUEST_LANGUAGE_ITA);
        }
```

File: src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java [hunk 3]
```diff
@@ -123,6 +123,181 @@ void shouldRetrieveFieldsDtoUsingExplicitParameters() {
                 .verifyComplete();
     }
 
+    @Test
+    void shouldRetrieveFieldsDtoUsingExplicitParametersWithLangExplicitSetIta() {
```
Comment (giovanniberti) [inline src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java:127]: Here you could use [`@ParameterizedTest`](https://www.baeldung.com/parameterized-tests-junit-5)

Comment (giovanniberti) [reply src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java:127]: Also: is there a test with `null` as the language parameter?

Comment (EmanueleBVtech) [reply src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java:127]: yes, most tests here do not expect language as this was added only in this PR

Comment (EmanueleBVtech) [reply src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java:127]: ParametrizedTest require JUnit 5 with is not currently included in the project.
We can take note of this and maybe open another PR specifically to improve the tests?

File: src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java (hunk 1 above)
Comment (EmanueleBVtech) [reply src/main/java/it/pagopa/ecommerce/commons/client/NpgClient.java]: done

File: src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java (hunk 3 above)
Comment (giovanniberti) [reply src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java:127]: I think we include JUnit 5 transitively, because we do already have some [parameterized tests](https://github.com/pagopa/pagopa-ecommerce-commons/blob/main/src/test/java/it/pagopa/ecommerce/commons/domain/BINTest.java#L24) 🤣
I would add them and include a new task in the backlog that we need to check versions' dependencies

Comment (EmanueleBVtech) [reply src/test/java/it/pagopa/ecommerce/commons/client/NpgClientTests.java]: Right. i was looking in the POM and got misleaded. I added them now. Thanks

**Review by giovanniberti — APPROVED** [review]
